scripts/examples/gn_silabs_example.sh

```diff
@@ -28,14 +28,23 @@ else
     CHIP_ROOT="$MATTER_ROOT"
 fi
 
+ACTIVATE_MATTER_ENV=true
+if [[ -z "${PW_ENVIRONMENT_ROOT}" ]]; then
+    echo "Using the bootstrapped pigweed ENV in Matter root"
+    PW_PATH="$CHIP_ROOT/.environment/cipd/packages/pigweed"
+else
+    echo "Using provided $PW_ENVIRONMENT_ROOT as Pigweed ENV root"
+    PW_PATH="$PW_ENVIRONMENT_ROOT"
+    ACTIVATE_MATTER_ENV=false
+fi
+
 set -x
 env
 USE_WIFI=false
 USE_DOCKER=false
 USE_GIT_SHA_FOR_VERSION=true
 USE_SLC=false
-GN_PATH=gn
-GN_PATH_PROVIDED=false
+GN_PATH="$PW_PATH/gn"
```

Comment:
should we check if this exists?

in a vscode instance for example, PW_ENVIRONMENT_ROOT will be set however without bootstrap nothing will be populated there. So it seems here we want bootstrap to be run somewhere, but this script to not be run under a bootstrapped environment after that. Is this true?

Reply:
In this case, there might be something else missing. I couldn't bootstrap a provided `PW_ENVIRONMENT_ROOT` from the matter root where this script is run. This PW_ENVIRONMENT_ROOT had to be pre-bootstrap.

I was under the impression the passed PW_ENVIRONMENT_ROOT was ready to use.

Reply:
PW_ENVIRONMENT_ROOT is the environment that pigweed should use at the time of running `bootstrap`. It is not guaranteed to be populated and it always starts as empty (probably not even an existing directory).